src/flow/core/data.rs

```diff
@@ -225,26 +247,26 @@ impl<'a> From<&'a str> for DataType {
     }
 }
 
+impl fmt::Debug for DataType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let enum_type = match *self {
+            DataType::None => "None",
+            DataType::Text(..) => "Text",
+            DataType::TinyText(..) => "TinyText",
+            DataType::Int(..) => "Int",
+            DataType::BigInt(..) => "BigInt",
+            DataType::Real(..) => "Real",
+            DataType::Timestamp(..) => "Timestamp",
+        };
+
+        // Output e.g. Int (5):
+        write!(f, "{} ({})", enum_type, self.format())
```

Comment:
I think it'd be better to implement `format` directly inside the match arms above. That way, you can directly use the debug formatters to output the values. For example:
```rust
match *self {
            DataType::TinyText(..) => {
                let text: Cow<str> = self.into();
                write!("TinyText({:?})", text)
            }
            // ...
        }
```
This has the advantage of "correctly" printing debug strings, so that, e.g., the string `I contain ' and "` appears as
```
TinyText("I contain ' and \"")
```

Reply:
Good idea - changed it in 5d6fad3. `Real` still makes use of `Display.fmt`'s logic though. Do you think that should be abstracted out somehow?

Also added a couple of test assertions for `Timestamp`.